pkg/server/node.go

```diff
@@ -651,17 +651,38 @@ func (n *Node) start(
 		return errors.Wrapf(err, "couldn't gossip descriptor for node %d", n.Descriptor.NodeID)
 	}
 
-	// Create stores from the engines that were already initialized.
-	for _, e := range state.initializedEngines {
-		s := kvserver.NewStore(ctx, n.storeCfg, e, &n.Descriptor)
-		if err := s.Start(workersCtx, n.stopper); err != nil {
-			return errors.Wrap(err, "failed to start store")
+	// Create stores in parallel from the engines that were already initialized. This uses a channel  to collect errors
+	// when starting stores. The channel is a buffered channel with the same size as the number of stores so worker
+	// go routines will never wait.
+	c := make(chan error, len(state.initializedEngines))
```

Comment:
nit: since this remains in scope, let's give it a proper name, e.g. `engineErrC`.